RequestService/RequestService.Handlers/GetAllJobsByFilterHandler.cs

```diff
@@ -0,0 +1,204 @@
+                {
+                    var location = locationDetails.First(x => x.Location == j.Location);
+
+                    if (location != null)
+                    {
+                        j.PostCode = location.Address.Postcode;
+                    }
+                }
+            }
+        }
+
+        public async Task<GetAllJobsByFilterResponse> Handle(GetAllJobsByFilterRequest request, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrEmpty(request.Postcode))
+            {
+                request.Postcode = HelpMyStreet.Utils.Utils.PostcodeFormatter.FormatPostcode(request.Postcode);
+
+                try
+                {
+                    var postcodeValid = await _addressService.IsValidPostcode(request.Postcode, cancellationToken);
+                }
+                catch (HttpRequestException)
+                {
+                    throw new PostCodeException();
+                }
+            }
+
+            if (request.ExcludeSiblingsOfJobsAllocatedToUserID.HasValue)
+            {
+                if (request.JobStatuses?.JobStatuses.Count() == 0 || !request.JobStatuses.JobStatuses.Contains(JobStatuses.Open))
+                {
+                    throw new InvalidFilterException();
+                }
+            }
+
+            GetAllJobsByFilterResponse result = new GetAllJobsByFilterResponse() { JobSummaries = new List<JobSummary>(), ShiftJobs = new List<ShiftJob>() };
+
+            if (request.Groups != null && request.Groups.Groups.Count == 0)
+            {
+                return result;
+            }
+
+            List<int> referringGroups = new List<int>();
+
+            if (request.ReferringGroupID.HasValue)
+            {
+                referringGroups.Add(request.ReferringGroupID.Value);
+
+                if (request.IncludeChildGroups)
+                {
+                    var childGroups = await _groupService.GetChildGroups(request.ReferringGroupID.Value);
+
+                    if (childGroups.ChildGroups.Count > 0)
+                    {
+                        referringGroups.AddRange(childGroups.ChildGroups.Select(x => x.GroupId).ToList());
+                    }
+
+                }
+            }
+
+            List<JobDTO> allJobs = _repository.GetAllJobs(request, referringGroups);
+
+            if (allJobs.Count == 0)
+                return result;
+
+            await AddPostCodeForLocations(allJobs);
+
+            allJobs = await _jobFilteringService.FilterAllJobs(
+                allJobs,
+                request.Postcode,
+                request.DistanceInMiles,
+                request.ActivitySpecificSupportDistancesInMiles,
+                cancellationToken);
+
+            if (request.ExcludeSiblingsOfJobsAllocatedToUserID.HasValue)
+            {
+                var userRequests = allJobs.Where(x => x.VolunteerUserID == request.ExcludeSiblingsOfJobsAllocatedToUserID.Value).Distinct(_jobBasicDedupe_EqualityComparer);
```

Comment:
allJobs is not _all_  jobs!  It's only the jobs that get through the filtering done in the stored proc.
I think here we need to make a separate call to the repo to get the jobs allocated to the user.  (Otherwise, for example, if this is a request for Open jobs then allJobs will only include Open jobs - and not any sibling jobs that the user may have accepted.)